src/lib/Form/Data/ContentTypeData.php

```diff
@@ -23,11 +22,8 @@ class ContentTypeData extends ContentTypeUpdateStruct implements NewnessCheckabl
      */
     use NewnessChecker;
 
-    /** @var \eZ\Publish\API\Repository\Values\ContentType\ContentTypeDraft */
-    protected $contentTypeDraft;
-
-    /** @var \EzSystems\EzPlatformAdminUi\Form\Data\FieldDefinitionData[] */
-    protected $fieldDefinitionsData = [];
+    /** @var \EzSystems\EzPlatformAdminUi\Form\Data\FieldDefinitionData[][] */
+    public $fieldDefinitionsData = [];
```

Comment:
This has potential for disaster. I've checked and updated tests that relied on the fact that the array was one dimensional. Once grouping is introduced it becomes a lot harder to follow.

If possible, both grouped and ungrouped variants should be reachable. For example, grouped variant could be returned from a new method. Grouping relies on `FieldDefinitionData` properties (`fieldGroup`, `identifier`) anyway, so it can be done on demand in my opinion.

Additionally, why is the property changed to `public` visibility?

Reply:
I've analyzed the Ibexa Commerce code and usage of this property. While I'm not okay with the fact that we're doing it like this (basically changing how the property, which already exposed, consists of) considering the alternative (introducing collection object) I'm ok with it.

It has to be public for the form component to interact with the array correctly when setting new values.